rotkehlchen/chain/ethereum/modules/thegraph/accountant.py

```diff
@@ -0,0 +1,79 @@
+from collections import defaultdict
+from collections.abc import Iterator
+from typing import TYPE_CHECKING
+
+from rotkehlchen.accounting.mixins.event import AccountingEventType
+from rotkehlchen.accounting.structures.evm_event import get_tx_event_type_identifier
+from rotkehlchen.accounting.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.chain.ethereum.modules.thegraph.constants import CPT_THEGRAPH
+from rotkehlchen.chain.evm.accounting.interfaces import ModuleAccountantInterface
+from rotkehlchen.chain.evm.accounting.structures import TxEventSettings
+from rotkehlchen.chain.evm.types import string_to_evm_address
+from rotkehlchen.constants import ZERO
+from rotkehlchen.fval import FVal
+
+if TYPE_CHECKING:
+    from rotkehlchen.accounting.pot import AccountingPot
+    from rotkehlchen.accounting.structures.evm_event import EvmEvent
+    from rotkehlchen.assets.asset import Asset
+    from rotkehlchen.types import ChecksumEvmAddress
+
+
+class ThegraphAccountant(ModuleAccountantInterface):
+    def reset(self) -> None:
+        self.assets_supplied: dict[tuple[ChecksumEvmAddress, Asset], FVal] = defaultdict(FVal)
+
+    def _process_deposit(
+            self,
+            pot: 'AccountingPot',  # pylint: disable=unused-argument
+            event: 'EvmEvent',
+            other_events: Iterator['EvmEvent'],  # pylint: disable=unused-argument
+    ) -> None:
+        self.assets_supplied[(string_to_evm_address(event.location_label), event.asset)] += event.balance.amount  # type: ignore[arg-type]  # location_label can't be None here  # noqa: E501
+
+    def _process_withdraw(
+            self,
+            pot: 'AccountingPot',
+            event: 'EvmEvent',
+            other_events: Iterator['EvmEvent'],  # pylint: disable=unused-argument
+    ) -> None:
+        key = (string_to_evm_address(event.location_label), event.asset)  # type: ignore[arg-type]  # location_label can't be None here
+        self.assets_supplied[key] -= event.balance.amount
```

Comment:
no need to repeat same message many times in the file.

```suggestion
        key = string_to_evm_address(event.location_label)  # type: ignore[arg-type]
        self.assets_supplied[key] -= event.balance.amount
```